### ISSUE TITLE
[V3 Bank] bank.withdraw/deposit_credits breaks mongodb with floats

### ISSUE THREAD

**Issue by officiallyemma** [Cog-Creators/Red-DiscordBot#1975]
# Other bugs

If you give bank.deposit_credits or bank.withdraw_credit a float instead of an int while using MongoDB, the users balance is stuck as a double. Because of this many cogs break because they expect an int, but get a float.

#### What were you trying to do?

I was trying to withdraw credits from a users account with

`await bank.withdraw_credits(ctx.author, X * 0.30)`

although due to a bug I accidentaly ran

```
await bank.withdraw_credits(ctx.author, X * 0.30)
await bank.deposit_credits(ctx.author, X * 0.30)
````

This is most likely unrelated to the bug, but I thought I should mention it anyway.

#### What were you expecting to happen?

I expected red to either throw an error or convert the float to an int 

#### What actually happened?

Red carried on, as if the value was an int. It changed the value type of the balance in mongodb from an int32 to a double. This breaks many other cogs because they expect an int but get a double.

#### How can we reproduce this issue?

1. run the following command

```
await bank.withdraw_credits(ctx.author, X * 0.30)
await bank.deposit_credits(ctx.author, X * 0.30)
```

2. done

**Comment by Tobotimus** [comment]
Hi! Thanks for reporting this issue. In my opinion it is best to throw an error when a float is passed to `withdraw/deposit_credits`, since a float value for credits doesn't really make any sense. We could implicitly convert it to int as you say, but I don't think it's hard for the cog creator to do that instead, and it saves having to account for implicit behaviour.

**Comment by officiallyemma** [comment]
+1 yeah I think throwing an error if the input isn't an int is the way to go.

**Comment by Tobotimus** [comment]
Resolved by #1976.
